### PR TITLE
Added etcd network latency analysis

### PR DESCRIPTION
**Pull request by seshachalam-yv**
**How to categorize this PR?**

/area documentation

**What this PR does / why we need it**:
This PR captures the etcd network latency analysis for single node and mult-node.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @seshachalam-yv for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: docs/etcd-network-latency.md [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+      | Number of keys | Value size | Number of connections | Number of clients | Target etcd server |  Average write QPS | Average latency per request |    zone    | server name |  Test name |
+      |:--------------:|:----------:|:---------------------:|:-----------------:|:------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+      |      10000     |     256    |           1           |         1         |       leader       |      1220.0520     |           0.815ms           | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+      |      10000     |     256    |           1           |         1         |       leader       |       586.545      |            1.74ms           | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+      |      10000     |     256    |           1           |         1         |       leader       |  554.0155654442634 |            1.8ms            | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+
+  * Benchmark tool loads key/value to `follower` with single client.
+    * `mn-etcd-sz` latency(`~2.2ms`) is **20% to 30% less than** `mn-etcd-mz`(`~2.7ms`).
+    * *Compare to `follower`, `leader` latency is less.*
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:56]:
```suggestion
    * *Compare to `follower`, `leader` has lower latency.*
```

File: docs/etcd-network-latency.md [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+      | Number of keys | Value size | Number of connections | Number of clients | Target etcd server |  Average write QPS | Average latency per request |    zone    | server name |  Test name |
+      |:--------------:|:----------:|:---------------------:|:-----------------:|:------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+      |      10000     |     256    |           1           |         1         |     follower-2     |       457.967      |            2.17ms           | eu-west-1a | etcd-main-2 | mn-etcd-sz |
+      |      10000     |     256    |           1           |         1         |     follower-2     |  345.6586129825796 |            2.89ms           | eu-west-1b | etcd-main-2 | mn-etcd-mz |
+
+  * Benchmark tool loads key/value to `leader` with multiple clients.
+    * `sn-etcd-sz` latency(`~78.3ms`) is **~10% greater than**  `mn-etcd-sz`(`~71.81ms`).
+    * `mn-etcd-sz` latency(`~71.81ms`) is less than `mn-etcd-mz`(`~72.5ms`) but the difference is not much.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:70]:
```suggestion
    * `mn-etcd-sz` latency(`~71.81ms`) is less than `mn-etcd-mz`(`~72.5ms`) but the difference is negligible.
```

File: docs/etcd-network-latency.md [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+      |:--------------:|:----------:|:---------------------:|:-----------------:|:------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+      |     100000     |     256    |          100          |        1000       |       leader       |      12638.905     |           78.32ms           | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+      |     100000     |     256    |          100          |        1000       |       leader       |      13789.248     |           71.81ms           | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+      |     100000     |     256    |          100          |        1000       |       leader       | 13728.446436395223 |            72.5ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |  
+
+
+
+  * Benchmark tool loads key/value to `follower` with multiple clients.
+    * `mn-etcd-sz` latency(`~69.8ms`) is **~5% greater than**  `mn-etcd-mz`(`~72.6ms`).
+    * *Compare to `leader`, `follower` latency is less*.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:82]:
```suggestion
    * *Compare to `leader`, `follower` has lower latency*. 
```

File: docs/etcd-network-latency.md [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+      |     100000     |     256    |          100          |        1000       |     follower-2     | 15750.409490407475 |            63.3ms           | eu-west-1b | etcd-main-2 | mn-etcd-mz |
+
+  </details>
+
+* 
+  <details>  
+  <summary>In this case benchmark tool tries to put key with random 1 MB value.</summary>
+
+  * Benchmark tool loads key/value to `leader` with single client.
+    * `sn-etcd-sz` latency(`~16.35ms`) is **~20% less than** `mn-etcd-sz`(`~20.64ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:101]:
```suggestion
    * `sn-etcd-sz` latency(`~16.35ms`) is **~20% lesser than** `mn-etcd-sz`(`~20.64ms`).
```

File: docs/etcd-network-latency.md [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+    * `mn-etcd-sz` latency(`~20.64ms`) is less than `mn-etcd-mz`(`~21.08ms`) but the difference is not much..
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:102]:
```suggestion
    * `mn-etcd-sz` latency(`~20.64ms`) is less than `mn-etcd-mz`(`~21.08ms`) but the difference is negligible..
```

File: docs/etcd-network-latency.md [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+      | Number of keys | Value size | Number of connections | Number of clients | Target etcd server |  Average write QPS | Average latency per request |    zone    | server name |  Test name |
+      |:--------------:|:----------:|:---------------------:|:-----------------:|:------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+      |      1000      |   1000000  |           1           |         1         |       leader       |       61.117       |           16.35ms           | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+      |      1000      |   1000000  |           1           |         1         |       leader       |       48.416       |           20.64ms           | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+      |      1000      |   1000000  |           1           |         1         |       leader       |  45.7517341664802  |           21.08ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+
+  * Benchmark tool loads key/value withto `follower` single client.
+    * `mn-etcd-sz` latency(`~23.10ms`) is **~10% greater than** `mn-etcd-mz`(`~21.8ms`).
+    * *Compare to `follower`, `leader` latency is less*.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:112]:
```suggestion
    * *Compare to `follower`, `leader` has lower latency*.
```

File: docs/etcd-network-latency.md [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+      |      1000      |   1000000  |           1           |         1         |     follower-2     |       40.0518      |           24.95ms           | eu-west-1a | etcd-main-2 | mn-etcd-sz |
+      |      1000      |   1000000  |           1           |         1         |     follower-2     |  43.28573155709838 |           23.09ms           | eu-west-1b | etcd-main-2 | mn-etcd-mz |
+      |      1000      |   1000000  |           1           |         1         |     follower-2     |        45.92       |           21.76ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+      |      1000      |   1000000  |           1           |         1         |     follower-2     |       35.5705      |            28.1ms           | eu-west-1b | etcd-main-2 | mn-etcd-mz |
+
+  * Benchmark tool loads key/value to `leader` with multiple clients.
+    * `sn-etcd-sz` latency(`~6.0375secs`) is **~30% greater than**  `mn-etcd-sz``~4.000secs`).
+    * `mn-etcd-sz` latency(`~4.000secs`) is less than `mn-etcd-mz`(`~ 4.09secs`) but the difference is not much.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:129]:
```suggestion
    * `mn-etcd-sz` latency(`~4.000secs`) is less than `mn-etcd-mz`(`~ 4.09secs`) but the difference is negligible.
```

File: docs/etcd-network-latency.md [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+      |:--------------:|:----------:|:---------------------:|:-----------------:|:------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+      |      1000      |   1000000  |          100          |        300        |       leader       |       55.373       |          6.0375secs         | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+      |      1000      |   1000000  |          100          |        300        |       leader       |       67.319       |          4.000secs          | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+      |      1000      |   1000000  |          100          |        300        |       leader       |  65.91914167957594 |           4.09secs          | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+
+  * Benchmark tool loads key/value to `follower` with multiple clients.
+    * *`mn-etcd-sz` latency(`~4.04secs`) is **~5% greater than** `mn-etcd-mz`(`~ 3.90secs`).*
+    * *Compare to `leader`, `follower` latency is less*.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:139]:
```suggestion
    * *Compare to `leader`, `follower` has lower latency*. 
```

File: docs/etcd-network-latency.md [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+### Latency analysis during Range requests to ETCD 
+
+* 
+  <details>
+  <summary>In this case benchmark tool tries to get specific key with random 256 bytes value.</summary>
+  
+  * Benchmark tool range requests to `leader` with single client.
+
+    * `sn-etcd-sz` latency(`~1.24ms`) is **~40% greater than** `mn-etcd-sz`(`~0.67ms`).
+    * `mn-etcd-sz` latency(`~0.67ms`) is  **~20% less than** `mn-etcd-mz`(`~0.85ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:199]:
```suggestion
    * `mn-etcd-sz` latency(`~0.67ms`) is  **~20% lesser than** `mn-etcd-mz`(`~0.85ms`).
```

File: docs/etcd-network-latency.md [hunk 10]
```diff
@@ -0,0 +1,467 @@
...
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |        10000       |     256    |           1           |         1         |       true      |      s      |        leader       |      1411.229      |            0.70ms           | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+    |        10000       |     256    |           1           |         1         |       true      |      s      |        leader       |      2033.131      |            0.35ms           | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+    |        10000       |     256    |           1           |         1         |       true      |      s      |        leader       | 2100.2426362012025 |            0.47ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+
+  * Benchmark tool range requests to `follower` with single client .
+    * `mn-etcd-sz` latency(`~1.3ms`) is  **~20% less than** `mn-etcd-mz`(`~1.6ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:216]:
```suggestion
    * `mn-etcd-sz` latency(`~1.3ms`) is  **~20% lesser than** `mn-etcd-mz`(`~1.6ms`).
```

File: docs/etcd-network-latency.md [hunk 11]
```diff
@@ -0,0 +1,467 @@
...
+    |        10000       |     256    |           1           |         1         |       true      |      s      |      follower-1     |      1823.631      |            0.54ms           | eu-west-1a | etcd-main-0 | mn-etcd-sz |
+    |        10000       |     256    |           1           |         1         |       true      |      s      |      follower-1     |       1442.6       |            0.69ms           | eu-west-1c | etcd-main-0 | mn-etcd-mz |
+    |        10000       |     256    |           1           |         1         |       true      |      s      |      follower-1     |       1416.39      |            0.70ms           | eu-west-1c | etcd-main-0 | mn-etcd-mz |
+    |        10000       |     256    |           1           |         1         |       true      |      s      |      follower-1     |      2077.449      |            0.47ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+  
+  * Benchmark tool range requests to `leader` with multiple client.
+    * `sn-etcd-sz` latency(`~84.66ms`) is **~20% greater than** `mn-etcd-sz`(`~73.95ms`).
+    * `mn-etcd-sz` latency(`~73.95ms`) is  **more or less equal** `mn-etcd-mz`(`~ 73.8ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:234]:
```suggestion
    * `mn-etcd-sz` latency(`~73.95ms`) is  **more or less equal to** `mn-etcd-mz`(`~ 73.8ms`).
```

File: docs/etcd-network-latency.md [hunk 12]
```diff
@@ -0,0 +1,467 @@
...
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |       100000       |     256    |          100          |        1000       |       true      |      l      |        leader       |      11775.721     |           84.66ms           | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+    |       100000       |     256    |          100          |        1000       |       true      |      l      |        leader       |     13446.9598     |           73.95ms           | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+    |       100000       |     256    |          100          |        1000       |       true      |      l      |        leader       |  13527.19810605353 |            73.8ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+    
+    * Compare to consistency `Linearizable`, `Serializable` is **~20% less** for all cases
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:242]:
```suggestion
    * Compare to consistency `Linearizable`, `Serializable` is **~20% lesser** for all cases
```

File: docs/etcd-network-latency.md [hunk 13]
```diff
@@ -0,0 +1,467 @@
...
+    * `sn-etcd-sz` latency(`~69.37ms`) is  **more or less equal** `mn-etcd-sz`(`~69.89ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:243]:
```suggestion
    * `sn-etcd-sz` latency(`~69.37ms`) is  **more or less equal to** `mn-etcd-sz`(`~69.89ms`).
```

File: docs/etcd-network-latency.md [hunk 14]
```diff
@@ -0,0 +1,467 @@
...
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |       100000       |     256    |          100          |        1000       |       true      |      s      |        leader       |     14334.9027     |           69.37ms           | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+    |       100000       |     256    |          100          |        1000       |       true      |      s      |        leader       |      14270.008     |           69.89ms           | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+    |       100000       |     256    |          100          |        1000       |       true      |      s      |        leader       | 14715.287354023869 |           67.63ms           | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+
+  * Benchmark tool range requests to `follower` with multiple client.
+    * `mn-etcd-sz` latency(`~60.69ms`) is **~20% less than** `mn-etcd-mz`(`~70.76ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:253]:
```suggestion
    * `mn-etcd-sz` latency(`~60.69ms`) is **~20% lesser than** `mn-etcd-mz`(`~70.76ms`).
```

File: docs/etcd-network-latency.md [hunk 15]
```diff
@@ -0,0 +1,467 @@
...
+    * Compare to  `leader`, `follower` read request latency is less.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:254]:
```suggestion
    * Compare to  `leader`, `follower` has lower read request latency.    
```

File: docs/etcd-network-latency.md [hunk 16]
```diff
@@ -0,0 +1,467 @@
...
+    *  Compare to `mn-etcd-mz` consistency `Linearizable`, `Serializable` is **~slightly less**.
+    *  
+    | Number of requests | Value size | Number of connections | Number of clients | sequential-keys | Consistency |  Target etcd server |  Average write QPS | Average latency per request |    zone    | server name |  Test name |
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |       100000       |     256    |          100          |        1000       |       true      |      s      |      follower-1     |      11582.438     |           86.09ms           | eu-west-1a | etcd-main-0 | mn-etcd-sz |
+    |       100000       |     256    |          100          |        1000       |       true      |      s      |      follower-1     |       15422.2      |            64.6ms           | eu-west-1c | etcd-main-0 | mn-etcd-mz |
+
+
+  * Benchmark tool range requests to `leader` all keys.
+    * `sn-etcd-sz` latency(`~678.77ms`) is **~5% slightly less than** `mn-etcd-sz`(`~697.29ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:273]:
```suggestion
    * `sn-etcd-sz` latency(`~678.77ms`) is **~5% slightly lesser than** `mn-etcd-sz`(`~697.29ms`).
```

File: docs/etcd-network-latency.md [hunk 17]
```diff
@@ -0,0 +1,467 @@
...
+    * `mn-etcd-sz` latency(`~697.29ms`) is less than `mn-etcd-mz`(`~701ms`) but the difference is not much.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:274]:
```suggestion
    * `mn-etcd-sz` latency(`~697.29ms`) is less than `mn-etcd-mz`(`~701ms`) but the difference is negligible.
```

File: docs/etcd-network-latency.md [hunk 18]
```diff
@@ -0,0 +1,467 @@
...
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |         20         |     256    |           2           |         5         |      false      |      l      |        leader       |       6.8875       |           678.77ms          | eu-west-1c | etcd-main-0 | sn-etcd-sz |
+    |         20         |     256    |           2           |         5         |      false      |      l      |        leader       |        6.720       |           697.29ms          | eu-west-1a | etcd-main-1 | mn-etcd-sz |
+    |         20         |     256    |           2           |         5         |      false      |      l      |        leader       |         6.7        |            701ms            | eu-west-1a | etcd-main-1 | mn-etcd-mz |
+
+    * * Compare to consistency `Linearizable`, `Serializable` is **~5% slightly higher** for all cases
+    * `sn-etcd-sz` latency(`~687.36ms`) is less than `mn-etcd-sz`(`~692.68ms`) but the difference is not much.
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:283]:
```suggestion
    * `sn-etcd-sz` latency(`~687.36ms`) is less than `mn-etcd-sz`(`~692.68ms`) but the difference is negligible.
```

File: docs/etcd-network-latency.md [hunk 19]
```diff
@@ -0,0 +1,467 @@
...
+    * `mn-etcd-sz` latency(`~692.68ms`) is **~5% slightly less than** `mn-etcd-mz`(`~735.7ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:284]:
```suggestion
    * `mn-etcd-sz` latency(`~692.68ms`) is **~5% slightly lesser than** `mn-etcd-mz`(`~735.7ms`).
```

File: docs/etcd-network-latency.md [hunk 20]
```diff
@@ -0,0 +1,467 @@
...
+  </details>
+
+* 
+  <details>
+
+  <summary>In this case benchmark tool tries to get specific key with random `1MB` value.</summary>
+  
+  * Benchmark tool range requests to `leader` with single client.
+
+    * `sn-etcd-sz` latency(`~5.96ms`) is **~5% less than** `mn-etcd-sz`(`~6.28ms`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:325]:
```suggestion
    * `sn-etcd-sz` latency(`~5.96ms`) is **~5% lesser than** `mn-etcd-sz`(`~6.28ms`).
```

File: docs/etcd-network-latency.md [hunk 21]
```diff
@@ -0,0 +1,467 @@
...
+    | Number of requests | Value size | Number of connections | Number of clients | sequential-keys | Consistency |  Target etcd server |  Average write QPS | Average latency per request |    zone    | server name |  Test name |
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |        1000        |   1000000  |           1           |         1         |       true      |      s      |      follower-1     |       170.918      |            5.84ms           | eu-west-1a | etcd-main-0 | mn-etcd-sz |
+    |        1000        |   1000000  |           1           |         1         |       true      |      s      |      follower-1     |       199.01       |            5.01ms           | eu-west-1c | etcd-main-0 | mn-etcd-mz |
+
+
+
+  * Benchmark tool range requests to `leader` with multiple clients.
+
+    * `sn-etcd-sz` latency(`~1.593secs`) is **~20% less than** `mn-etcd-sz`(`~1.974secs`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:364]:
```suggestion
    * `sn-etcd-sz` latency(`~1.593secs`) is **~20% lesser than** `mn-etcd-sz`(`~1.974secs`).
```

File: docs/etcd-network-latency.md [hunk 22]
```diff
@@ -0,0 +1,467 @@
...
+    | Number of requests | Value size | Number of connections | Number of clients | sequential-keys | Consistency |  Target etcd server |  Average write QPS | Average latency per request |    zone    | server name |  Test name |
+    |:------------------:|:----------:|:---------------------:|:-----------------:|:---------------:|:-----------:|:-------------------:|:------------------:|:---------------------------:|:----------:|:-----------:|:----------:|
+    |        1000        |   1000000  |          100          |        500        |       true      |      s      |      follower-2     |       183.320      |          2.2810secs         | eu-west-1a | etcd-main-2 | mn-etcd-sz |
+    |        1000        |   1000000  |          100          |        500        |       true      |      s      |      follower-2     |       195.40       |          2.164secs          | eu-west-1b | etcd-main-2 | mn-etcd-mz |
+
+
+
+  * Benchmark tool range requests to `leader` all keys.
+
+    * `sn-etcd-sz` latency(`~8.993secs`) is **~3% slightly less than** `mn-etcd-sz`(`~9.236secs`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:417]:
```suggestion
    * `sn-etcd-sz` latency(`~8.993secs`) is **~3% slightly lower than** `mn-etcd-sz`(`~9.236secs`).
```

File: docs/etcd-network-latency.md [hunk 23]
```diff
@@ -0,0 +1,467 @@
...
+    * `mn-etcd-sz` latency(`~9.236secs`) is  **~2% slightly less than** `mn-etcd-mz`(`~9.100secs`).
```
Comment (aaronfern) [inline docs/etcd-network-latency.md:418]:
```suggestion
    * `mn-etcd-sz` latency(`~9.236secs`) is  **~2% slightly lower than** `mn-etcd-mz`(`~9.100secs`).
```

**Review by aaronfern — CHANGES_REQUESTED** [review]
Thanks for the PR @seshachalam-yv!
Just a few minor suggestions from me

**Comment by seshachalam-yv** [comment]
Thanks for review @aaronfern

**Review by aaronfern — APPROVED** [review]
/lgtm

**Pull request merged**